wp-content/plugins/core/src/Templates/Content/Single/Post.php

```diff
@@ -31,13 +32,19 @@ public function defer_get_content() {
 		return apply_filters( 'the_content', get_the_content() );
 	}
 
-	protected function get_featured_image() {
+	protected function get_featured_image(): string {
+		$image_id = get_post_thumbnail_id();
+
+		if ( empty( $image_id ) ) {
```

Comment:
Dumb q, does the image component not handle this on its own? Would it make sense for it to do so?

Reply:
Not a dumb question. I actually went back & forth on this myself.

The Image component (any of the components, actually) doesn't prevent you from rendering their markup without any data. I think this makes sense architecturally, and I didn't want to deviate from that construct in this instance.

Additionally, the image component can take an `IMG_URL` property instead of the `IMG_ID` and render a strait URL to an image file. Thus, we would have to check & bail only if both inputs were empty (minor change, of course).

Reply:
I also don't mind it being here - it allows us to bail before we do any processing at all within the Image component. I would be super-shocked if that was ever any sort of bottleneck, but I'm always a fan of not instantiating classes/processing if we don't need to :)